jax/_src/tree_util.py

```diff
@@ -204,7 +204,7 @@ def tree_multimap(*args, **kwargs):
 # TODO(mattjj,phawkins): consider removing this function
 def _process_pytree(process_node, tree):
   leaves, treedef = pytree.flatten(tree)
-  return treedef.walk(process_node, None, leaves), treedef
+  return treedef.walk(lambda node, data: process_node(node), None, leaves), treedef
 
```

Comment:
Adapt to old API, could just choose now to delete this function, or propagate the API to the currently unknown list of clients of this function.